Sources/KeyboardShortcuts/SwiftUI+Extensions.swift

```diff
@@ -0,0 +1,123 @@
+// Created by Neil Clayton on 17/07/2024.
+// Using ideas by mbenoukaiss, https://github.com/sindresorhus/KeyboardShortcuts/issues/101
+
+
+import Foundation
+import SwiftUI
+import KeyboardShortcuts
+import Carbon
+
+// Provides a SwiftUI like wrapper func, that feels the same as the normal SwiftUI keyboardShortcut
+@available(macOS 11.0, *)
+extension View {
+	@ViewBuilder
+	public func keyboardShortcut(_ shortcutName: KeyboardShortcuts.Name) -> some View {
+		KeyboardShortcutView(shortcutName: shortcutName) {
+			self
+		}
+	}
+}
+
+// Holds the state of the shortcut, and changes that state when the shortcut changes
+// This causes the related NSMenuItem to also update (yipeee)
+@available(macOS 11.0, *)
+struct KeyboardShortcutView<Content: View>: View {
+	@State private var shortcutName: KeyboardShortcuts.Name
+	@State private var shortcut: KeyboardShortcuts.Shortcut?
+
+	private var content: () -> Content
+
+	init(shortcutName: KeyboardShortcuts.Name, content: @escaping () -> Content) {
+		self.shortcutName = shortcutName
+		self.shortcut = KeyboardShortcuts.getShortcut(for: shortcutName)
+		self.content = content
+	}
+
+	@ViewBuilder
+	var shortcutBody: some View {
+		if let shortcut, let keyEquivalent = shortcut.toKeyEquivalent() {
+			content()
+					.keyboardShortcut(keyEquivalent, modifiers: shortcut.toEventModifiers())
+		} else {
+			content()
+		}
+	}
+
+	var body: some View {
+		shortcutBody
+				// Called only when the shortcut is updated
+				.onReceive(NotificationCenter.default.publisher(for: .shortcutByNameDidChange)) { notification in
+					if let name = notification.userInfo?["name"] as? KeyboardShortcuts.Name, name == shortcutName {
+						let current = KeyboardShortcuts.getShortcut(for: name)
+						// this updates the shortcut state locally, refreshing the View, thus updating the SwiftUI menu item
+						// It's also fine if it is nil (which happens when you set the shortcut, in RecorderCocoa).
+						// See the comment on becomeFirstResponder (in short: so that you can reassign the SAME keypress to a shortcut, without it whining that it's already in use)
+						print("Shortcut \(shortcutName) updated to: \(current?.description ?? "nil")")
+						shortcut = current
+					}
+				}
+	}
+}
+
+@available(macOS 11.0, *)
+extension KeyboardShortcuts.Shortcut {
+	func toKeyEquivalent() -> KeyEquivalent? {
```

Comment:
There is similar code in https://github.com/sindresorhus/KeyboardShortcuts/blob/2e5f15581fefb821d4b366e57d817be8bf12aa58/Sources/KeyboardShortcuts/Shortcut.swift#L247 It should be unified.